src/transactionsFactories/tokenManagementTransactionIntentsFactory.spec.ts

```diff
@@ -218,4 +219,158 @@ describe("test token management transactions factory", () => {
         assert.equal(transaction.receiver, grace.address.toString());
         assert.equal(transaction.value, 0n);
     });
+
+    it("should create 'Transaction' for modifying royalties", () => {
+        const transaction = tokenManagementFactory.createTransactionForModifyingRoyalties({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+            royalties: 1234n,
+        });
+
+        assert.deepEqual(transaction.data, Buffer.from("ESDTModifyRoyalties@544553542d313233343536@01@04d2"));
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, grace.address.toString());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60125000n);
+    });
+
+    it("should create 'Transaction' for setting new uris", () => {
+        const transaction = tokenManagementFactory.createTransactionForSettingNewUris({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+            uris: ["firstURI", "secondURI"],
+        });
+
+        assert.deepEqual(
+            transaction.data,
+            Buffer.from("ESDTSetNewURIs@544553542d313233343536@01@6669727374555249@7365636f6e64555249"),
+        );
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, grace.address.toString());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60164000n);
+    });
+
+    it("should create 'Transaction' for modifying creator", () => {
+        const transaction = tokenManagementFactory.createTransactionForModifyingCreator({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+        });
+
+        assert.deepEqual(transaction.data, Buffer.from("ESDTModifyCreator@544553542d313233343536@01"));
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, grace.address.toString());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60114500n);
+    });
+
+    it("should create 'Transaction' for updating metadata", () => {
+        const transaction = tokenManagementFactory.createTransactionForUpdatingMetadata({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+            tokenName: "Test",
+            royalties: 1234n,
+            hash: "abba",
+            attributes: Buffer.from("test"),
+            uris: ["firstURI", "secondURI"],
```

Comment:
So that there is no ambiguity regarding what gets changed, we can prefix all the changed fields with "new". E.g. we'll have `tokenNonce`, but we'll have `newTokenName`.

Additionally, maybe receive a `Token` class directly instead of `tokenIdentifier` & `tokenNonce`? Or maybe not, to keep consistency with existing methods of this factory?